vggish_audio_encoder.py

```diff
@@ -44,33 +45,35 @@ def __init__(self,
         self.post_processor = Postprocessor(self.pca_path)
 
     @requests
-    def encode(self, docs: Optional[DocumentArray], parameters: dict, *args, **kwargs) -> Any:
-
+    def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
+        """
+        Compute embeddings and store them in the `docs` array.
+
+        :param docs: documents sent to the encoder. The docs must have `text`.
+            By default, the input `text` must be a `list` of `str`.
+        :param parameters: dictionary to define the `traversal_paths` and the `batch_size`. For example,
+               `parameters={'traversal_paths': ['r'], 'batch_size': 10}`.
+        :param kwargs: Additional key value arguments.
+        :return:
+        """
         if docs:
-            cleaned_document_array = self._get_input_data(docs, parameters)
+            cleaned_document_array = get_docs_batch_generator(
+                docs,
+                traversal_path=parameters.get('traversal_paths', self.default_traversal_paths),
+                batch_size=1,
+                needs_attr='blob'
+            )
             self._create_embeddings(cleaned_document_array)
 
-    def _get_input_data(self, docs: DocumentArray, parameters: dict):
-        """Create a filtered set of Documents to iterate over."""
-
-        traversal_paths = parameters.get('traversal_paths', self.default_traversal_paths)
-
-        # traverse thought all documents which have to be processed
-        flat_docs = docs.traverse_flat(traversal_paths)
-
-        # filter out documents without images
-        filtered_docs = DocumentArray([doc for doc in flat_docs if doc.blob is not None])
-
-        return filtered_docs
-
     def _create_embeddings(self, filtered_docs: Iterable):
         """Update the documents with the embeddings generated by VGGISH"""
 
         for d in filtered_docs:
             # Vggish broadcasts across different length audios, not batches
-            [embedding] = self.sess.run([self.embedding_tensor], feed_dict={self.feature_tensor: d.blob})
+            print('###', len(d))
+            [embedding] = self.sess.run([self.embedding_tensor], feed_dict={self.feature_tensor: d[0].blob})
```

Comment:
same here d[0]????

Reply:
yes, it looks weird. The reason is that the generator yields batches, i.e. `DocumentArray`.
But vggish does not broadcast batches so we need to use batch_size=1 and take d[0].

Reply:
Maybe for executors that don't support batching the function is not that useful.
Or I can rename `d` to `doc_batch` and add a comment for clarification?

What do you think?

Reply:
if there is no batch being used there is no point to use the function

Reply:
agree

Reply:
https://github.com/jina-ai/executor-audio-VGGishEncoder/pull/7/commits/de64fa42a400f6034ed6866931542e30ac905ecd

Reply:
Removed the function. Not closing PR because  docstrings were missing and are worth adding